x-pack/legacy/plugins/snapshot_restore/public/app/index.tsx

```diff
@@ -34,20 +36,23 @@ export const useAppDependencies = () => {
 
 const getAppProviders = (deps: AppDependencies) => {
   const {
-    i18n: { Context: I18nContext },
+    i18n: { Context: I18nContext, FormattedMessage },
   } = deps.core;
 
   // Create App dependencies context and get its provider
   const AppDependenciesProvider = setAppDependencies(deps);
 
   return ({ children }: { children: ReactNode }) => (
-    <I18nContext>
-      <HashRouter>
-        <AppDependenciesProvider value={deps}>
-          <AppStateProvider value={useReducer(reducer, initialState)}>{children}</AppStateProvider>
-        </AppDependenciesProvider>
-      </HashRouter>
-    </I18nContext>
+    <AuthorizationProvider
+      permissionEndpoint={httpService.addBasePath(`${API_BASE_PATH}permissions`)}
```

Comment:
Maybe, instead of having each app creating their handler for the permissions, we could have a single handler in our shared plugin, that would expose something like:

```
GET /<plugin_name>/authorization/permissions?app=snapshot_restore
```

We could then remove the need of this `permissionEndpoint` prop here.

Reply:
a single handler may get complex since I think the needs of each app will differ from each other. I like this approach for now

Reply:
I don't think it would be complex. 75% of the handler code is generic and very little specific to the app. We would simply need to create 1 extension file / app for what is specific.
The benefit would be that we could then simply inject the AuthroizatioProvider anywhere, add an extension to the shared plugin "/privileges" endpoint handler and be done.

We can do that as a second pass 😊